--- joe/Makefile.make	Thu Dec 11 16:56:01 1997
+++ joe/Makefile	Thu Dec 11 17:02:19 1997
@@ -9,15 +9,15 @@
 # to go and where you want the man page
 # to go:
 
-WHEREJOE = /usr/bin
-WHERERC = /usr/lib/joe
-WHEREMAN = /usr/man/man1
+WHEREJOE = $(TOPDIR)/usr/bin
+WHERERC = $(TOPDIR)/usr/lib/joe
+WHEREMAN = $(TOPDIR)/usr/man/man1
 
 # If you want to use TERMINFO, you have to set
 # the following variable to 1.  Also you have to
 # include some additional libraries- see below.
 
-TERMINFO = 0
+TERMINFO = 1
 
 # You may also have to add some additional
 # defines to get the include files to work
@@ -26,12 +26,10 @@
 # for some HPUX systems, you need to add:  -D_HPUX_SOURCE
 
 # C compiler options: make's built-in rules use this variable
-
 CFLAGS = $(RPM_OPT_FLAGS)
 
 # C compiler to use: make's built-in rules use this variable
-
-CC = cc
+CC = gcc
 
 # You may have to include some extra libraries
 # for some systems
@@ -68,53 +66,30 @@
 	ln joe jpico
 
 # All object files depend on config.h
-
 $(OBJS): config.h
 
 # How to make config.h
-
 config.h:
 	$(CC) conf.c -o conf
 	./conf $(WHERERC) $(TERMINFO)
 
 # How to make termidx
-
 termidx: termidx.o
 	$(CC) $(CFLAGS) -o termidx termidx.o
 
 # Install proceedure
-
 install: joe termidx
-	strip joe
-	strip termidx
-	if [ ! -d $(WHEREJOE) ]; then mkdir $(WHEREJOE); chmod a+rx $(WHEREJOE); fi
-	rm -f $(WHEREJOE)/joe $(WHEREJOE)/jmacs $(WHEREJOE)/jstar $(WHEREJOE)/jpico $(WHEREJOE)/rjoe 
-	mv joe $(WHEREJOE)
-	ln $(WHEREJOE)/joe $(WHEREJOE)/jmacs
-	ln $(WHEREJOE)/joe $(WHEREJOE)/jstar
-	ln $(WHEREJOE)/joe $(WHEREJOE)/rjoe
-	ln $(WHEREJOE)/joe $(WHEREJOE)/jpico
-	if [ ! -d $(WHERERC) ]; then mkdir $(WHERERC); chmod a+rx $(WHERERC); fi
-	rm -f $(WHERERC)/joerc $(WHERERC)/jmacsrc $(WHERERC)/jstarrc $(WHERERC)/jpicorc $(WHERERC)/rjoerc $(WHEREMAN)/joe.1
-	cp joerc $(WHERERC)
-	cp jmacsrc $(WHERERC)
-	cp jstarrc $(WHERERC)
-	cp rjoerc $(WHERERC)
-	cp jpicorc $(WHERERC)
-	cp joe.1 $(WHEREMAN)
-	chmod a+x $(WHEREJOE)/joe
-	chmod a+x $(WHEREJOE)/jmacs
-	chmod a+x $(WHEREJOE)/jstar
-	chmod a+x $(WHEREJOE)/rjoe
-	chmod a+x $(WHEREJOE)/jpico
-	chmod a+r $(WHERERC)/joerc
-	chmod a+r $(WHERERC)/jmacsrc
-	chmod a+r $(WHERERC)/jstarrc
-	chmod a+r $(WHERERC)/rjoerc
-	chmod a+r $(WHERERC)/jpicorc
-	chmod a+r $(WHEREMAN)/joe.1
+	[ ! -d $(WHEREJOE) ] && mkdir -p $(WHEREJOE) 
+	install -s -m0555 joe $(WHEREJOE)
+	ln -s joe $(WHEREJOE)/jmacs
+	ln -s joe $(WHEREJOE)/jstar
+	ln -s joe $(WHEREJOE)/rjoe
+	ln -s joe $(WHEREJOE)/jpico
+	[ ! -d $(WHERERC) ] &&	mkdir -p $(WHERERC)
+	install -m0644 joerc jmacsrc jstarrc rjoerc jpicorc $(WHERERC)
+	[ ! -d $(WHEREMAN) ] && mkdir -p $(WHEREMAN)
+	install -m 0644 joe.1 $(WHEREMAN)
 
 # Cleanup proceedure
-
 clean:
 	rm -f $(OBJS) termidx.o conf conf.o config.h
