--- joe/vfile.c.foo	Fri Mar 31 14:59:13 2000
+++ joe/vfile.c	Fri Mar 31 15:00:03 2000
@@ -255,7 +255,7 @@
 {
 struct stat buf;
 VFILE *new=(VFILE *)malloc(sizeof(VFILE));
-new->name=vsncpy(NULL,0,sz(name));
+new->name=strdup(name);
 new->fd=open(name,O_RDWR);
 if(!new->fd)
  {
@@ -288,7 +288,7 @@
  {
  if(vfile->flags) unlink(vfile->name);
  else vflshf(vfile);
- vsrm(vfile->name);
+ free(vfile->name);
  }
 if(vfile->fd) close(vfile->fd);
 free(deque(VFILE,link,vfile));
