--- joe-2.9.7/Makefile.am~	Wed Nov  7 15:54:40 2001
+++ joe-2.9.7/Makefile.am	Sat Nov 10 19:43:30 2001
@@ -26,11 +26,10 @@
 termidx_SOURCES = termidx.c
 
 install-exec-hook:
-	rm -f $(bindir)/jmacs $(bindir)/jstar $(bindir)/rjoe $(bindir)/jpico
-	for i in jmacs jstar rjoe jpico ; do $(LN_S) joe $(bindir)/$$i ; done
+	for i in jmacs jstar rjoe jpico ; do $(LN_S) joe $(DESTDIR)$(bindir)/$$i ; done
 
 uninstall-local:
-	rm -f $(bindir)/jmacs $(bindir)/jstar $(bindir)/rjoe $(bindir)/jpico
+	rm -f $(DESTDIR)$(bindir)/jmacs $(DESTDIR)$(bindir)/jstar $(DESTDIR)$(bindir)/rjoe $(DESTDIR)$(bindir)/jpico
 
 distclean-local:
 	rm -f *~
