diff -urN utils.c utils.c
--- utils.c	2004-04-25 23:10:36.077137480 +0200
+++ utils.c	2004-04-25 23:11:27.019393080 +0200
@@ -35,7 +35,7 @@
 /*
  * Define function isblank(c)
  *	!!! code which uses isblank() assumes tested char is evaluated
- *	only once, so it musn't be a macro
+ *	only once, so it mustn't be a macro
  */
 /* GNU is blank does not work properly for wide characters */
 
@@ -85,7 +85,7 @@
 	if (wide)
 		return (iswalnum(c) || (c == 95));
 	else
-		return (isalnum(c) || (c == 95));
+	    	return (c >= 0 && c < 256 && (isalnum(c) || (c == 95)));
 }
 
 /* Versions of 'read' and 'write' which automatically retry when interrupted */
