--- joe/b.c	Sat Nov 18 13:57:08 2000
+++ joe-/b.c	Sat Nov 18 13:56:21 2000
@@ -19,6 +19,7 @@
 #include <stdio.h>
 #ifndef __MSDOS__
 #include <pwd.h>
+#include <fcntl.h>
 #endif
 #include <errno.h>
 
@@ -1990,7 +1991,20 @@
  {
  long tim=time(0);
  B *b;
- FILE *f=fopen("DEADJOE","a");
+ FILE *f;
+ int fd,i;
+ char buf[100];
+ 
+ fd=open("DEADJOE",O_WRONLY|O_CREAT|O_EXCL);
+ if (fd==-1) {
+  for(i=1;i<1000;i++) {
+   snprintf(buf,99,"DEADJOE.%d",i);
+   fd=open(buf,O_WRONLY|O_CREAT|O_EXCL);
+   if(fd!=-1)break;
+  }
+ }
+ if(fd==-1)return;
+ f=fdopen(fd,"w");
  fchmod(fileno(f), 0600);
  fprintf(f,"\n*** Modified files in JOE when it aborted on %s",ctime(&tim));
  if(sig) fprintf(f,"*** JOE was aborted by signal %d\n",sig);
