diff -urNp -x '*.orig' joe-4.6.org/rc/jmacsrc.in joe-4.6/rc/jmacsrc.in
--- joe-4.6.org/rc/jmacsrc.in	2018-01-10 17:28:34.000000000 +0100
+++ joe-4.6/rc/jmacsrc.in	2026-03-03 00:14:39.953285294 +0100
@@ -27,7 +27,7 @@
  -right nn	Amount to jump right when scrolling is necessary (-1 for 1/4 width)
 -marking	Text between ^KB and cursor is highlighted
 -lightoff	Turn off highlighting after block copy or move
- -asis		Characters 128 - 255 shown as-is
+-asis		Characters 128 - 255 shown as-is
  -force		Force final newline when files are saved
  -nolocks	If you don't want file locks to be used
  -nomodcheck	Disable periodic checking of disk file newer than buffer
diff -urNp -x '*.orig' joe-4.6.org/rc/joerc.in joe-4.6/rc/joerc.in
--- joe-4.6.org/rc/joerc.in	2018-01-10 17:28:34.000000000 +0100
+++ joe-4.6/rc/joerc.in	2026-03-03 00:14:39.953663405 +0100
@@ -27,7 +27,7 @@
 
  -marking	Text between ^KB and cursor is highlighted (use with -lightoff)
 
- -asis		Characters 128 - 255 shown as-is
+-asis		Characters 128 - 255 shown as-is
 
  -force		Force final newline when files are saved
 
diff -urNp -x '*.orig' joe-4.6.org/rc/jpicorc.in joe-4.6/rc/jpicorc.in
--- joe-4.6.org/rc/jpicorc.in	2018-01-10 17:28:34.000000000 +0100
+++ joe-4.6/rc/jpicorc.in	2026-03-03 00:14:39.953969707 +0100
@@ -30,7 +30,7 @@
 
 -lightoff	Turn off highlighting after block copy or move
 
- -asis		Characters 128 - 255 shown as-is
+-asis		Characters 128 - 255 shown as-is
 
 -force		Force final newline when files are saved
 
diff -urNp -x '*.orig' joe-4.6.org/rc/jstarrc.in joe-4.6/rc/jstarrc.in
--- joe-4.6.org/rc/jstarrc.in	2018-01-10 17:28:34.000000000 +0100
+++ joe-4.6/rc/jstarrc.in	2026-03-03 00:14:39.954278806 +0100
@@ -24,7 +24,7 @@
  -left nn	Amount to jump left when scrolling is necessary (-1 for 1/4 width)
  -right nn	Amount to jump right when scrolling is necessary (-1 for 1/4 width)
  -marking	Text between ^KB and cursor is highlighted (use with -lightoff)
- -asis		Characters 128 - 255 shown as-is
+-asis		Characters 128 - 255 shown as-is
  -force		Force final newline when files are saved
  -nolocks	If you don't want file locks to be used
  -nomodcheck	Disable periodic checking of disk file newer than buffer
diff -urNp -x '*.orig' joe-4.6.org/rc/rjoerc.in joe-4.6/rc/rjoerc.in
--- joe-4.6.org/rc/rjoerc.in	2018-01-10 17:28:34.000000000 +0100
+++ joe-4.6/rc/rjoerc.in	2026-03-03 00:14:39.954518086 +0100
@@ -28,7 +28,7 @@
 
  -marking	Text between ^KB and cursor is highlighted (use with -lightoff)
 
- -asis		Characters 128 - 255 shown as-is
+-asis		Characters 128 - 255 shown as-is
 
  -force		Force final newline when files are saved
 
